File: job-offer/src/main/java/com/oxygensend/joboffer/infrastructure/storage/StorageProperties.java
package com.oxygensend.joboffer.infrastructure.storage;

import jakarta.validation.constraints.NotEmpty;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

@Validated
@ConfigurationProperties(prefix = "storage")
record StorageProperties(@NotEmpty String attachmentsLocation) {

}

File: static-data/src/main/kotlin/com/oxygensend/staticdata/infrastructure/storage/StorageProperties.kt
package com.oxygensend.staticdata.infrastructure.storage

import jakarta.validation.constraints.NotEmpty
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties(prefix = "storage")
data class StorageProperties(@field:NotEmpty val aboutUsImageDir: String?)
